tools/site_admin/copy_pages.php

```diff
@@ -383,6 +386,33 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
         die( "You can't copy a project into itself." );
     }
 
+    $charsuites_not_in_to = array_udiff(
+        $project_obj["from"]->get_charsuites(FALSE),
+        $project_obj["to"]->get_charsuites(FALSE),
+        function ($a, $b) { return $a == $b ? 0 : 1; }
+    );
+    if ($charsuites_not_in_to)
+    {
+        echo "<p class='warning'>" . _("Character Suite mismatch") . "</p>";
+        echo "<p>" . _("The following character suites are in the source project but not in the destination project. Ensure the copied pages do not use these or adjust the destination project character suites accordingly.") . "</p>";
+        echo "<ul>";
+        foreach($charsuites_not_in_to as $charsuite)
+        {
+            echo "<li>" . $charsuite->name . "</li>";
```

Comment:
Html safe would be nice here just in case

Reply:
Character suite names are defined in the code and while unlikely to contain beyond alphanumeric, dashes, and underscores technically could. I'll get this in.